packages/lesswrong/server/migrations/migrationUtils.ts

```diff
@@ -298,6 +298,12 @@ export async function forEachDocumentBatchInCollection({collection, batchSize, f
   loadFactor?: number
 })
 {
+  // As described in the docstring, we need to be able to query on the _id.
+  // Without this check, someone trying to use _id in the filter would overwrite
+  // this function's query and find themselves with an infinite loop.
+  if (filter && '_id' in filter) {
```

Comment:
Assuming you're trying to just check for the presence of the property, then this is a nice use of the `in` operator, kudos. At first I thought it was important to check if `_id` had a set value, and was going to ping you, but I think that first thought was wrong. Noting this here in case I was actually right and you need to do a value check.

Reply:
It also seems like you could get this for free by just inverting the destructure order of the filter criteria on L319, i.e. `{ ...filter, _id: { $gt: lastId } }`, though I guess maybe that's harder to reason about in the future.

Reply:
1st Comment: You could imagine checking for truthiness of _id with `filter?._id`, but even a falsey _id will overwrite.

2nd Comment: But in that case I'm allowing someone to pass in an _id field to the filter which would then get overruled.